Bowling/Bowling/Score.cs

```diff
@@ -0,0 +1,109 @@
+﻿using NUnit.Framework.Constraints;
+using NUnit.Framework.Internal.Execution;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+
+namespace BowlingApp
+{
+    public class Score
+    {
+        public const int frameNumberTotal = 10;
+        public const int frameNumberWithBonusRoll = 10;
+
+        public List<Frame> CreateScoreCard(string bowlingMarks)
```

Comment:
This is a large function, please re-compose it into several more functions.

Reply:
I rewrote some of the logic in this to get the placeCount dependency removed.  I will let you take a look at this one on the next review and see if you still think its too large.

Reply:
I think we can further simplify this (you may need to see my next comment first about `OrganizeBowlingMarksByFrame` for this to work).

```
public IEnumerable<Frame> CreateScoreCard(string bowlingMarks)
{
    var frameNumber = 1;
    var bowlingMarksByFrame = OrganizeBowlingMarksByFrame(bowlingMarks);

    foreach (var marks in bowlingMarksByFrame)
    {
        if (!OnLastFrame(frameNumber))
        {
            var frameNoBonusRoll = new FrameNoBonusRollFactory();
            yield frameNoBonusRoll.Create(frameNumber, marks.ToString());
        }
        else
        {
            var frameWithBonusRoll = new FrameWithBonusRollFactory();
            yield frameWithBonusRoll.Create(marks.ToString());
        }

        frameNumber++;
    }
}
```

Reply:
Updated CreateScoreCard to add yield return functionality.  Thanks Drew for your assistance in getting this to work correctly!